src/main/java/fr/openent/competences/helpers/DevoirControllerHelper.java

```diff
@@ -58,90 +63,93 @@ public void handle(Either<String, JsonObject> event) {
                             .put("structureId", devoir.getStructureId())
                             .put("groupId", devoir.getGroupId())
                             .put("userId", user.getUserId());
-                    eb.send(Competences.VIESCO_BUS_ADDRESS, action, getReplyHandler(devoirWithId, shareService, user, request));
+                    eb.request(Competences.VIESCO_BUS_ADDRESS, action, getReplyHandler(devoirWithId, user, devoirsService, request));
                 } else {
                     badRequest(request);
                 }
             }
         };
     }
 
-    public static Handler<Either<String, JsonArray>> getDuplicationDevoirHandler(UserInfos user, ShareService shareService, HttpServerRequest request, EventBus eb) {
-        return new Handler<Either<String, JsonArray>>() {
-            @Override
-            public void handle(Either<String, JsonArray> event) {
-                if (event.isRight()) {
-                    final JsonArray devoirs = event.right().getValue();
-                    ArrayList<Future<JsonObject>> futures = new ArrayList<>();
-                    for(Object devoirO : devoirs) {
-                        Promise<JsonObject> promise = Promise.promise();
-                        futures.add(promise.future());
-                        // recuperation des professeurs que l'utilisateur connecté remplacent
-                        JsonObject devoirJO = (JsonObject)devoirO;
-                        Devoir devoir = new Devoir(devoirJO.getJsonObject("devoir"));
-                        JsonObject action = new JsonObject()
-                                .put("action", "multiTeaching.getIdMultiTeachers")
-                                .put("subjectId", devoir.getSubjectId())
-                                .put("structureId", devoir.getStructureId())
-                                .put("groupId", devoir.getGroupId())
-                                .put("userId", user.getUserId());
-                        eb.request(Competences.VIESCO_BUS_ADDRESS, action, getReplyHandler(devoirJO, shareService, user, promise));
-                    }
-                    FutureHelper.all(futures)
-                            .onSuccess(success -> request.response().setStatusCode(200).end())
-                            .onFailure(failure -> badRequest(request, failure.getMessage()));
-
-                } else {
-                    badRequest(request);
+    public static Handler<Either<String, JsonArray>>
+    getDuplicationDevoirHandler(UserInfos user, DevoirService devoirService, HttpServerRequest request, EventBus eb) {
+        return event -> {
+            if (event.isRight()) {
+                final JsonArray devoirs = event.right().getValue().getJsonArray(1);
+                final JsonArray groupes_rel = event.right().getValue().getJsonArray(3);
```

Comment:
risque de index out of bound, une petite vérif s'impose je pense

Reply:
P'tite question au passage, il y a un endroit expliquant les différente valeurs de ton tableau ?
Car actuellement d'un point de vu lisibilité de code, ce n'est pas super pratique de lire des éléments avec des index.

Reply:
```suggestion
                List<JsonObject> devoirs =event.right().getValue().getJsonArray(1, new JsonArray()).getList()
                List<JsonObject> groupes_rel =event.right().getValue().getJsonArray(3, new JsonArray()).getList()
```
ça te facilite le typage pour la suite